cleverhans/model_abs.py

```diff
@@ -0,0 +1,73 @@
+from abc import ABCMeta
+from keras.models import Model
+
+
+class ModelAbstraction:
+    """
+    An abstract interface for a model that exposes the required functionalities
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, model):
+        """
+        :param model: A function that takes a symbolic input and returns the
+                      symbolic output for the model's predictions.
+        """
+        raise NotImplementedError('Model abstraction not implementated.')
```

Comment:
I think we actually want this to just be `pass`. We probably want subclasses to call this constructor. In the future, we might have the base class do some generic work in the constructor (register the model with some name server, or something like that---just speculating). If we put an exception in the constructor, no one will call it, and we'll have to change all the subclass code if we ever depend on the constructor being called.